lib/api/include/irods/check_auth_credentials.h

```diff
@@ -0,0 +1,134 @@
+#ifndef IRODS_CHECK_AUTH_CREDENTIALS_H
+#define IRODS_CHECK_AUTH_CREDENTIALS_H
+
+/// \file
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_check_auth_credentials.
+///
+/// \since 4.3.1
+typedef struct CheckAuthCredentialsInput // NOLINT(modernize-use-using)
+{
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
+    char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
+    char zone[64];
+
+    /// The expected obfuscated password for the user of interest.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed 250 - 1 bytes
```

Comment:
I know we're ignoring magic numbers in this section, but why 250?

Reply:
That is the max size defined in the database for passwords.

I couldn't think of any other value to use here because it wasn't clear what the max size for an obfuscated password is.

Reply:
Okay, mostly just curious. This can be resolved.

Reply:
Resolving https://github.com/irods/irods/issues/6297 will help here. So there's a path forward.